kong/tools/request_aware_table.lua

```diff
@@ -0,0 +1,122 @@
+--- NOTE: tool is designed to assist with **detecting** request contamination
+-- issues on CI, during test runs. It does not offer security safeguards.
+
+local table_new = require "table.new"
+local table_clear = require "table.clear"
+
+local debug_mode   = kong.configuration.log_level == "debug"
+local error        = error
+local rawset       = rawset
+local setmetatable = setmetatable
+local get_phase    = ngx.get_phase
+local var          = ngx.var
+local log          = ngx.log
+
+
+local NGX_VAR_PHASES = {
+  set           = true,
+  rewrite       = true,
+  access        = true,
+  content       = true,
+  header_filter = true,
+  body_filter   = true,
+  log           = true,
+  balancer      = true,
+}
+
+-- Check if access is allowed for table, based on the request ID
+local function enforce_sequential_access(table)
+  if not NGX_VAR_PHASES[get_phase()] then
+    -- not in a request context: allow access and reset allowed request ID
```

Comment:
I wonder if we could put that comment inside code like this:

```lua
local NGX_REQUEST_CONTEXT_PHASES = {
  set        = true,
  rewrite    = true
  -- etc
}

local function is_outside_request_context(phase)
  return not NGX_REQUEST_CONTEXT_PHASES[phase]
end

local function enforce_sequential_access(table)
  if is_outside_request_context(get_phase()) then
     -- etc
```

wdyt?

Reply:
actually it would be more appropriate to describe this as "not in a phase that can access ngx.var", which I think is self explanatory given the constant's name. I'm going to shorten the comment instead.